src/engine/handle-event.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Challenge,
+  Event,
+  EventType,
+  ResultEvent,
+  RunGameState,
+  TravelBingoRules,
+} from 'data/interfaces'
+
+import { handleChallengeCompleted } from './handle-challenge-completed'
+import { handleFinish } from './handle-finish'
+import { handleStart } from './handle-start'
+
+export function handleEvent(
+  event: Event,
+  state: RunGameState,
+  challenges: Challenge[][],
+  rules: TravelBingoRules,
+): ResultEvent[] {
+  const eventHandlers: {
+    [key in EventType]: (
+      event: Event,
+      state: RunGameState,
+      rules: TravelBingoRules,
+    ) => ResultEvent[]
+  } = {
+    [EventType.Empty]: () => [],
+    [EventType.Bingo]: () => [],
```

Comment:
This might be a stupid question but isn't bingo handled by the `handleChallengeCompleted` function? 🤔

Reply:
That's a good question! And makes me think about a refactoring!

The point is: we have `Event` and `ResultEvent` 
- `Event` are the "input" that we manually insert in the data files 
- `ResultEvent` extends `Event`, it's the "output" of this calculation. `ResultEvent` is enriched with more data and can have different types

For instance, one `Event` with type `ChallengeCompleted` can generate two `ResultEvent`, one `ChallengeCompleted` and other `Bingo`

The refactoring I'm thinking is create two different enums, let me try that.

Reply:
This approach is a little trick, I think. If you have any different suggestion, I would love to hear it!

Reply:
I needed another commit 😞 
https://github.com/geo-quest/travel-bingo/pull/47/commits/81d6d6adfb2db3ec5d241947688622d1628bff37

Reply:
Last one 🤣 https://github.com/geo-quest/travel-bingo/pull/47/commits/2aad85a88bfd22fed2a0761d2578fe61f34315a9

Reply:
Ahh gotcha, makes sense now 🙏

Reply:
No i think the way you implemented is pretty good, I love it 👍